src/Cake.Frosting/Internal/Extensions/FrostingTaskExtensions.cs

```diff
@@ -58,5 +68,39 @@ private static bool IsConvertableTo(this Type type, Type other)
         {
             return other == type || other.IsAssignableFrom(type);
         }
+
+        private static bool IsAsyncFrostingTask(this IFrostingTask task)
```

Comment:
If you know a better way of checking whether a type inherits an open generic type, I'm all 👂  &mdash; this looks like a lot of code.